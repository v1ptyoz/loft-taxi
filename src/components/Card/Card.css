.card {
  width: 350px;
  height: 180px;
  background: #FFFFFF;
  box-shadow: 0px 5px 20px 2px rgba(0, 0, 0, 0.1);
  border-radius: 10px;
  margin-top: 20px;
  padding: 18px 16px 16px 28px;
  display: flex;
  flex-direction: column;
}

.card__field {
  position: relative;
  font-size: 12px;
  display: flex;
}

.card__field--date {
  height: 33px;
  align-items: center;
  justify-content: flex-end;
}

.card__field--date::before {
  content: '';
  position: absolute;
  left: 0;
  top: 0;
  display: block;
  width: 33px;
  height: 33px;
  background: url("./card__logo.png");
}

.card__field--number {
  font-size: 21px;
  flex: 1;
  align-items: center;
}

.card__field--cvc {
  font-size: 18px;
  height: 28px;
  align-items: center;
  justify-content: center;
}

.card__field--cvc::before {
  content: '';
  position: absolute;
  left: 0;
  display: block;
  width: 30px;
  height: 26px;
  background: url("./card__chip.png") no-repeat;
}

.card__field--cvc::after {
  content: '';
  position: absolute;
  right: 0;
  display: block;
  width: 46px;
  height: 28px;
  background: url("./card__system.png") no-repeat;
}