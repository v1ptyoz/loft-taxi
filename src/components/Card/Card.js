import "./Card.css";

export function Card(props) {
  return (
    <div className="card">
      <div className="card__field card__field--date">05/08</div>
      <div className="card__field card__field--number">1234 5678 9012 3456 44</div>
      <div className="card__field card__field--cvc">123</div>
    </div>
  )
}