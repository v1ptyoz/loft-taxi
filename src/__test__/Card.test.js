import {Card} from "../components/Card/Card";
import renderer from "react-test-renderer";

describe("Button component", () => {
  it("Render normal via snapshot", () => {
    const card = renderer.create(<Card expiryDate="28/11" cardNumber="1234 5678" cvc="555" />)
    expect(card).toMatchSnapshot();
  })
})