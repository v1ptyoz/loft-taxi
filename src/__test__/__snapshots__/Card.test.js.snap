// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Render normal via snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card__field card__field--date"
  >
    28/11
  </div>
  <div
    className="card__field card__field--number"
  >
    1234 5678
  </div>
  <div
    className="card__field card__field--cvc"
  >
    555
  </div>
</div>
`;
